src/Datadog.Trace.ClrProfiler.Managed/Integrations/AspNet/HttpHeadersCollection.cs

```diff
@@ -0,0 +1,45 @@
+#if NETFRAMEWORK
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Datadog.Trace.Headers;
+
+namespace Datadog.Trace.ClrProfiler.Integrations.AspNet
+{
+    internal readonly struct HttpHeadersCollection : IHeadersCollection
```

Comment:
IRequestHeaders is a proxy (struct) so I think we can make this type to use the struct + constraint trick to avoid the allocation when boxing the struct to IRequestHeaders

Reply:
I'm still new to constrained virt calls so please excuse me if this is a silly question, but does this accomplish the struct + constraint trick? Or does this type need to be changed to have one generic type parameter `T` and then constrain that generic type parameter like `where T : IRequestHeaders`?

Reply:
And I'm new to both "constrained virt calls" _and_ the "struct + constraint trick" 😄 Is there anywhere I can read more about both? 🙂